src/optimizer/core/src/transform.rs

```diff
@@ -287,16 +295,124 @@ impl<'a> Fold for HookTransform<'a> {
 
     fn fold_fn_decl(&mut self, node: ast::FnDecl) -> ast::FnDecl {
         self.stack_ctxt.push(node.ident.sym.to_string());
+        self.decl_stack.push(vec![]);
         let o = node.fold_children_with(self);
         self.stack_ctxt.pop();
+        self.decl_stack.pop();
+
+        o
+    }
+
+    fn fold_arrow_expr(&mut self, node: ast::ArrowExpr) -> ast::ArrowExpr {
+        self.decl_stack.push(vec![]);
+        let current_scope = self.decl_stack.last_mut().unwrap();
```

Comment:
If really don't want to deal with `None`/`Err` cases here, at least we could replace this with an `expect`
```suggestion
        let current_scope = self.decl_stack.last_mut().expect("Declaration stack empty!");
```
So if a panic ever happens, at least we have some quick text/context on the precondition that failed (specially when ppl start using this in the wild with crazy code).